docs/api.rst

```diff
@@ -11505,3 +11505,216 @@ Handling user notes
+
+
+Custom Assets
+================
+
+.. http:post:: /api/(version)/assets/custom
+
+   Doing a POST on this endpoint will return all the custom assets present in the database using the filter parameters.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      POST /api/1/assets/custom HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {"name": "land", "custom_asset_type": "real estate"}
+
+   :reqjson int[optional] limit: This signifies the limit of records to return as per the `sql spec <https://www.sqlite.org/lang_select.html#limitoffset>`__.
+   :reqjson int[optional] offset: This signifies the offset from which to start the return of records per the `sql spec <https://www.sqlite.org/lang_select.html#limitoffset>`__.
+   :reqjson string[optional] name: The name of the custom asset to be used as a filter.
+   :reqjson string[optional] identifier: The identifier of the custom asset to be used as a filter.
+   :reqjson int[optional] started: The creation timestamp of the custom asset. This is optional.
+   :reqjson str[optional] swapped_for: The asset identifier it was swapped for. This is optional.
+   :reqjson string[optional] custom_asset_type: The type of the custom asset to be used as a filter.
+   :reqjson list[string] order_by_attributes: This is the list of attributes of the custom asset by which to order the results. By default we sort using ``name``.
+   :reqjson list[bool] ascending: Should the order be ascending? This is the default. If set to false, it will be on descending order.
+
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {
+          "result": {
+                "entries": [{
+                        "identifier": "00fe5f97-882b-42e5-b0e0-39ebd3a99156'",
+                        "name": "House",
+                        "notes": "Apartment purchase",
+                        "custom_asset_type": "real estate",
+                        "started": null,
+                        "swapped_for": null,
+                    },
+                    {
+                        "identifier": "333c248f-7c64-41f4-833b-2fad96c4ea6b",
+                        "name": "Ferrari",
+                        "notes": "Exotic car inheritance from lineage",
+                        "custom_asset_types": "vehicles",
+                        "started": null,
+                        "swapped_for": null,
+                }],
+                "entries_found": 2,
+                "entries_total": 2,
+            },
+          "message": ""
+      }
+
+   :resjson object result: An array of objects representing custom assets entries.
+   :resjson str identifier: The unique identifier of the custom asset.
+   :resjson str name: The name of the custom asset.
+   :resjson str notes: The notes used as a description of the custom asset. This field can be null.
```

Comment:
```suggestion
   :resjson str notes[optional]: The notes used as a description of the custom asset. This field can be null.
```

Reply:
no it's not optional like this. This we use only for request. This is a result. It would be there, but would be null.